7-feature_store/1-setup_feast.ipynb

```diff
@@ -75,16 +104,68 @@
+    "### 🏗️ How does Feast work?\n",
+    "\n",
+    "Feast organizes features into three key components:\n",
+    "- **📜 Registry:** A metadata store that keeps track of all feature definitions, sources, and entities.\n",
+    "- **📂 Offline Store:** A long-term storage system that holds historical feature data for training models.\n",
+    "- **⚡ Online Store:** A low-latency store optimized for real-time feature retrieval during inference."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {},
    "outputs": [],
+   "source": [
+    "import yaml\n",
+    "# Pretty-print the YAML configuration\n",
+    "print(yaml.dump(fs_config_yaml, default_flow_style=False, sort_keys=False, indent=2))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As mentioned earlier, Feast uses the `feature_store.yaml` file to [configure the Feature Store](https://docs.feast.dev/reference/feature-repository/feature-store-yaml#overview). If you examine the YAML configuration, you will recognize the three key components we just described.\n",
+    "\n",
+    "In our setup:\n",
+    "- The **Registry** and **Online Store** are configured to use a **PostgreSQL database**.\n",
+    "- The **Offline Store** is configured as a **file-based storage system**, which can be linked to an S3 bucket (in the `feature.py`, check the music_source that uses a s3 bucket as FileStore).\n",
+    "\n",
+    "This configuration ensures that features are efficiently stored, tracked, and served for both training and inference."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 🎯 Features and Feature Values\n",
+    "\n",
+    "In machine learning, **features** are the key pieces of data used as input signals for predictive models. In the context of a dataset, we distinguish between:\n",
+    "\n",
+    "- **Feature:** A complete column in the dataset that represents a measurable property (e.g., \"energy\" or \"speechiness\" of a song).\n",
+    "- **Feature Value:** A single data point from that feature column (e.g., the \"energy\" value for a specific song).\n",
+    "\n",
+    "Simply put, features provide the structured information that models use to make predictions. In our case, the **energy** and **speechiness** of a song are examples of features that could help determine whether a track becomes a hit."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 🚀 Applying Feast\n",
+    "\n",
+    "Before using our features, we first need to **apply** them. This step registers all the feature definitions inside the `feature_repo` with our **Feast registry**.\n",
+    "\n",
+    "In our setup, the registry is stored in a **PostgreSQL database**, which acts as a central metadata store. By applying Feast, we ensure that all feature definitions are properly cataloged and ready to be materialized for offline storage or served in real-time."
```

Comment:
Same here, they wouldn't be materialized for offline storage, but rather ready to be "fetched from offline storage" (or something similar), and "materialized for real-time serving".